### PR TITLE
doc: rgw server-side encryption and barbican

### PR DESCRIPTION
**Pull request by cbodley**
documentation to accompany https://github.com/ceph/ceph/pull/11049, based on a google doc from @aclamk 

a preview is available here: http://docs.ceph.com/docs/wip-rgw-encryption-doc/radosgw/encryption/

TODO:
- [X] add new config variables to `radosgw/config-ref.rst`
- [x] get sign-off from @aclamk to use the image in rgw-encryption-barbican.png
- [X] revisit exception: do the operations require ssl?
- [X] revisit exception: does HEAD request require SSE-C decryption key?

### CODE REVIEW

**Comment by cbodley** [comment]
@aclamk could you please help review? here are preview links:

encryption: http://docs.ceph.com/docs/wip-rgw-encryption-doc/radosgw/encryption/
barbican: http://docs.ceph.com/docs/wip-rgw-encryption-doc/radosgw/barbican/

also, can I add your `Signed-off-by` to the commit to say that it's okay for us to use the barbican image you provided?

**Comment by aclamk** [comment]
@cbodley How can I sign-off for rgw-encryption-barbican.png ? I want to do this, but I dont know how.

**Comment by aclamk** [comment]
@cbodley I think documentation is ok.

**Comment by cbodley** [comment]
thanks @aclamk! i added a `Signed-off-by: Adam Kupczyk <akupczyk@mirantis.com>` to the commit message

**Comment by cbodley** [comment]
jenkins test this please

File: doc/radosgw/encryption.rst [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+This is implemented in S3 according to the `Amazon SSE-C`_ specification.
+
+As all key management is handled by the client, no special configuration is
+needed to support this encryption mode.
+
+Key Management Service
+======================
+
+This mode allows keys to be stored in a secure key management service, and
```
Comment (adamemerson) [inline doc/radosgw/encryption.rst]: Delete the comma before and here. (And in this case joins two predicates sharing a subject, not two independent clauses.)

**Comment by cbodley** [comment]
thanks @adamemerson, fixed

**Comment by cbodley** [comment]
rebased

**Comment by cbodley** [comment]
jenkins test this please

**Pull request merged**